x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_rule_alerts_aggregations.ts

```diff
@@ -214,35 +214,45 @@ export async function fetchRuleAlertsAggByTimeRange({
         (
           acc: AlertChartData[],
           dayAlerts: {
-            key_as_string: string;
+            key: number;
             doc_count: number;
             alertStatus: {
               buckets: Array<{
-                key: string;
+                key: 'active' | 'recovered';
                 doc_count: number;
               }>;
             };
           }
         ) => {
           // We are adding this to each day to construct the 30 days bars (background bar) when there is no data for a given day or to show the delta today alerts/total alerts.
           const totalDayAlerts = {
-            date: dayAlerts.key_as_string,
-            count: totalAlerts,
+            date: dayAlerts.key,
+            count: totalAlerts === 0 ? 1 : totalAlerts,
             status: 'total',
           };
 
           if (dayAlerts.doc_count > 0) {
+            const localAlertChartData = acc;
             // If there are alerts in this day, we construct the chart data.
-            return [
-              ...acc,
-              ...dayAlerts.alertStatus.buckets.map((alert) => ({
-                date: dayAlerts.key_as_string,
+            let countOfRecoveredActiveAlerts = 0;
+            dayAlerts.alertStatus.buckets.forEach((alert) => {
+              countOfRecoveredActiveAlerts += alert.doc_count;
+              localAlertChartData.push({
+                date: dayAlerts.key,
                 count: alert.doc_count,
                 status: alert.key,
-              })),
-            ];
+              });
+            });
+            if (totalAlerts - countOfRecoveredActiveAlerts > 0) {
+              localAlertChartData.push({
+                date: dayAlerts.key,
+                count: totalAlerts - countOfRecoveredActiveAlerts,
```

Comment:
Nit: It should go with previous nit

```suggestion
                count: deltaAlertsCount,
```